src/approaches/oracle_approach.py

```diff
@@ -1201,3 +1211,172 @@ def _get_repeated_nextto_gt_nsrts() -> Set[NSRT]:
+                        r, [env.object_to_ig_object(o_i) for o_i in o]))
+                nsrts.add(nsrt)
+
+        elif base_option_name == "Grasp":
+            assert len(option_arg_type_names) == 1
+            target_obj_type_name = option_arg_type_names[0]
+            target_obj_type = type_name_to_type[target_obj_type_name]
+            target_obj = Variable("?targ", target_obj_type)
+
+            # Pick from ontop something
+            for surf_obj_type in sorted(env.types):
+                surf_obj = Variable("?surf", surf_obj_type)
+                parameters = [target_obj, agent_obj, surf_obj]
+                option_vars = [target_obj]
+                handempty = _get_lifted_atom("handempty", [])
+                targ_reachable = _get_lifted_atom("reachable",
+                                                  [target_obj, agent_obj])
+                targ_holding = _get_lifted_atom("holding", [target_obj])
+                preconditions = {handempty, targ_reachable}
+                add_effects = {targ_holding}
+                delete_effects = {handempty}
+                nsrt = NSRT(
+                    f"{option.name}-{next(op_name_count_pick)}",
+                    parameters,
+                    preconditions,
+                    add_effects,
+                    delete_effects,
+                    set(),
+                    option,
+                    option_vars,
+                    lambda s, r, o: grasp_obj_param_sampler(r),
+                )
+                nsrts.add(nsrt)
+
+        elif base_option_name == "PlaceOnTop":
+            assert len(option_arg_type_names) == 1
+            surf_obj_type_name = option_arg_type_names[0]
+            surf_obj_type = type_name_to_type[surf_obj_type_name]
+            surf_obj = Variable("?surf", surf_obj_type)
+
+            # We need to place the object we're holding!
+            for held_obj_types in sorted(env.types):
+                held_obj = Variable("?held", held_obj_types)
+                parameters = [held_obj, agent_obj, surf_obj]
+                option_vars = [surf_obj]
+                handempty = _get_lifted_atom("handempty", [])
+                held_holding = _get_lifted_atom("holding", [held_obj])
+                surf_reachable = _get_lifted_atom("reachable",
+                                                  [surf_obj, agent_obj])
+                ontop = _get_lifted_atom("ontop", [held_obj, surf_obj])
+                preconditions = {held_holding, surf_reachable}
+                add_effects = {ontop, handempty}
+                delete_effects = {held_holding}
+                nsrt = NSRT(
+                    f"{option.name}-{next(op_name_count_place)}",
+                    parameters,
+                    preconditions,
+                    add_effects,
+                    delete_effects,
+                    set(),
+                    option,
+                    option_vars,
+                    lambda s, r, o:  # type: ignore
```

Comment:
why do we need this type: ignore and the one in the line below?

Reply:
turns out we can get rid of it!